homeworks/08-cats-catcher/src/App/index.tsx

```diff
@@ -0,0 +1,79 @@
+import { FC, useState, useEffect, useRef } from "react";
+import { of, interval, fromEvent, combineLatest } from "rxjs";
+import { map, filter } from "rxjs/operators";
+import random from "lodash/random";
+
+import { ReactComponent as WindowIcon } from "assets/icons/window.svg";
+import { ReactComponent as CatIcon } from "assets/icons/kitty.svg";
+import { Container, IconWrapper, ControllersContainer } from "./App.styled";
+import { useGenerateWindowsPositions } from "./generateFiled";
+
+const App: FC = () => {
+  const catRef = useRef<HTMLSpanElement>(null);
+  const resetButtonRef = useRef<HTMLButtonElement>(null);
+  const { positions, onReset } = useGenerateWindowsPositions();
+  const [catPosition, setCatPosition] = useState<null | number[]>(null);
+  const [score, setScore] = useState(0);
+
+  useEffect(() => {
+    const positions$ = of(positions);
+    const interval$ = interval(1000);
+
+    const cat$ = combineLatest([positions$, interval$])
+      .pipe(
+        filter(([pos]) => {
```

Comment:
As you need to get the first element (windows positions) every time, will be nice to reorganise this code something like that:
```javascript
const catPosition$ = of(positions)
  .pipe(
    filter(Boolean),
    map((pos) => pos[random(0, 9)])
  )

const cat$ = interval(1000)
  .pipe(
    switchMap(() => catPosition$)
  )
```